src/consensus/aft/raft.h

```diff
@@ -2104,31 +2115,55 @@ namespace aft
         sort(match.begin(), match.end());
         auto confirmed = match.at((match.size() - 1) / 2);
 
-        if (confirmed < new_commit_idx)
+        if (
+          !new_agreement_index.has_value() ||
+          confirmed < new_agreement_index.value())
         {
-          new_commit_idx = confirmed;
+          new_agreement_index = confirmed;
         }
       }
-      RAFT_DEBUG_FMT(
-        "In update_commit, new_commit_idx: {}, "
-        "last_idx: {}",
-        new_commit_idx,
-        state->last_idx);
 
-      if (new_commit_idx != std::numeric_limits<Index>::max())
+      if (new_agreement_index.has_value())
       {
-        state->watermark_idx = new_commit_idx;
-      }
+        const auto new_commit_idx =
+          find_best_committable_index(new_agreement_index.value());
 
-      if (get_commit_watermark_idx() > state->last_idx)
-      {
-        throw std::logic_error(
-          "Followers appear to have later match indices than leader");
-      }
+        if (new_commit_idx.has_value())
+        {
+          if (new_commit_idx.value() > state->last_idx)
```

Comment:
This seems particularly impossible now that find_best_committable_index() is applied, but always good to check.